test_integration/01_deployment/01_deploy.js

```diff
@@ -139,4 +146,29 @@ describe('Deploy', async () => {
             ...auxiliaryAddresses,
         };
     });
+
+    it('active gateway ', async () => {
+        await shared.origin.addContract('EIP20Gateway');
```

Comment:
It's added twice now. See line 72. Maybe it is clearer to add it at the end of the deploy phase? together with the cogateway?

Reply:
How do I get contract instance to activate gateway? 🤔 
Should it be moved to deploy gateway and cogateway.

Reply:
If you move `await shared.origin.addContract('EIP20Gateway');` from `after` to the end of `it('correctly deploys Gateway and CoGateway')`, then you can access it here using `const gateway = shared.origin.contracts.EIP20Gateway;`. `shared` keeps it across tests and the integration tests are meant to be run all or none.
It might make sense to move all `shared.origin.addContract()` calls from after to the ends of their respective tests so that it is consistent.